### PR TITLE
[Doc] update k8s operator release note

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/ecosystem_release/spark_connector.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ displayed_sidebar: "English"
 | 1.1.1           | 3.2, 3.3, or 3.4 | 2.5 and later | 8    | 2.12  |
 | 1.1.0           | 3.2, 3.3, or 3.4 | 2.5 and later | 8    | 2.12  |
 
-## **Release note**
+## Release note
```
Comment (amber-create) [inline docs/en/ecosystem_release/spark_connector.md:31]: I guess it's better to use the plural form "Release notes"

File: docs/zh/ecosystem_release/starrocks-kubernetes-operator.md [hunk 2]
```diff
@@ -41,8 +41,62 @@ StarRocks 提供的 Operator 用于在 Kubernetes 环境中部署 StarRocks 集
...
 ## 发布记录
 
+### 1.9
+
+#### 1.9.1
+
+**功能改进**
+
+- **[Helm Chart]** 当 `logStorageSize` 设置为 `0` 时，operator 不会为 log storage 创建 PersistentVolumeClaim（PVC）。[#398](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/398)
+- **[Operator]** operator 会检查 `storageVolumes` 中 `mountPath` 和 `name` 的值是否重复时。如果存在重复的值，则会返回报错提示。[#388](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/388)
```
Comment (amber-create) [inline docs/zh/ecosystem_release/starrocks-kubernetes-operator.md:51]: 删掉”时“

File: docs/zh/ecosystem_release/starrocks-kubernetes-operator.md [hunk 3]
```diff
@@ -41,8 +41,62 @@ StarRocks 提供的 Operator 用于在 Kubernetes 环境中部署 StarRocks 集
...
+- **[Operator]** FE 节点的数量不能缩减到 1。[#394](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/394)
+- **[Operator]**  支持 merge 多个 values yaml 文件中定义`feEnvVars`、`beEnvVars` 和 `cnEnvVars` 的值。[#396](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/396)
```
Comment (amber-create) [inline docs/zh/ecosystem_release/starrocks-kubernetes-operator.md:53]: image
多余空格

File: docs/zh/ecosystem_release/starrocks-kubernetes-operator.md [hunk 4]
```diff
@@ -41,8 +41,62 @@ StarRocks 提供的 Operator 用于在 Kubernetes 环境中部署 StarRocks 集
...
+- **[Operator]** 在 StarRocksCluster CRD 中添加了 `spec.containers.securityContext.capabilities`，以自定义容器的 Linux 权限。[#404](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/404)
+
+**缺陷修复**
+
+已修复以下问题：
+
+- **[Operator]** 支持更新 `service` 中的 `annotations` 字段。[#402](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/402) [#399](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/399)
+- **[Operator]** 使用 patch 而不是 update 的方式来修改 statefulset 和 deployment。这可以解决启用 CN + HPA 时升级 CN 会导致所有 CN pods 被终止并重新启动的问题。[#397](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/397)
+- **[Operator]**  使用 patch 而不是 update 的方式来修改 service object。这样可以避免 operator 覆盖写入对  service object 的修改，例如，在使用 Kubernetes 云提供商时， 该 Kubernetes 云提供商修改了 service object。[#387](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/387)
```
Comment (amber-create) [inline docs/zh/ecosystem_release/starrocks-kubernetes-operator.md:62]: image
多余空格

Comment (amber-create) [reply docs/zh/ecosystem_release/starrocks-kubernetes-operator.md:62]: image
也有多余空格

File: docs/zh/ecosystem_release/starrocks-kubernetes-operator.md [hunk 5]
```diff
@@ -41,8 +41,62 @@ StarRocks 提供的 Operator 用于在 Kubernetes 环境中部署 StarRocks 集
...
+- **[Helm Chart]** 已更正  [starrockscluster.yaml](https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/helm-charts/charts/kube-starrocks/charts/starrocks/templates/starrockscluster.yaml) 中 storageSpec 中的拼写错误。 #385
```
Comment (amber-create) [inline docs/zh/ecosystem_release/starrocks-kubernetes-operator.md:63]: image
多余空格

File: docs/zh/ecosystem_release/starrocks-kubernetes-operator.md [hunk 6]
```diff
@@ -41,8 +41,62 @@ StarRocks 提供的 Operator 用于在 Kubernetes 环境中部署 StarRocks 集
...
+
+#### 1.9.0
+
+**新增特性**
+
+- 新增 StarRocksWarehouse CRD 以支持 StarRocks Warehouse。注意，StarRocks Warehouse 目前是 StarRocks 企业版的功能。[#323](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/323)
+
+**功能改进**
+
+- 在 StarRocksCluster CRD 中的  中添加了 `status.reason` 字段。如果在部署集群过程中 subcontroller 的 apply 操作失败时，您可以执行 `kubectl get starrockscluster <name_of_the_starrocks_cluster_object> -oyaml`, 在返回结果中查看 `status.reason` 字段显示的错误日志。[#359](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/359)
```
Comment (amber-create) [inline docs/zh/ecosystem_release/starrocks-kubernetes-operator.md:73]: image
多余空格

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 7]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
 
-## 1.8
+### 1.9
 
-### 1.8.6
+#### 1.9.1
+
+**Improvements**
+
+- **[Helm Chart]** The operator will not create PersistentVolumeClaim (PVC) for log storage when you set `logStorageSize` to `0`.  [#398](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/398)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:49]: when you set `logStorageSize` to `0` -> when/if when `logStorageSize` is set to `0`

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 8]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
+- **[Operator]** The operator can detect the values of`mountPath` and `name` in `storageVolumes` are duplicated. An error is returned when duplicate values exist. [#388](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/388)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:50]: The operator can detect the values of`mountPath` and `name` in `storageVolumes` are duplicated. -> The operator can detect whether the values of`mountPath` and `name` in `storageVolumes` are duplicated.

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 9]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
+- **[Operator]** The number of FE nodes can not be scaled down to 1. [#394](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/394)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:51]: can not -> cannot

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 10]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
+- **[Operator]** Add `spec.containers.securityContext.capabilities` in the StarRocksCluster CRD to customize the Linux capabilities of containers. [#404](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/404)
+
+**Bug Fixes**
+
+Fixed the following issues:
+
+- **[Operator]** The `annotations` field in `service` can be updated. [#402](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/402) [#399](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/399)
+- **[Operator]** The modification of statefulset and deployment is patched instead of updated. It solves the problem that upgrading CN will cause all CN pods to be terminated and restarted when CN and HPA are enabled. [#397](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/397)
+- **[Operator]** The modification of service object is patched instead of updated. It avoids the operator to overwrite the modification, for example, when a Kubernetes Cloud provider is used and the service object is modified by that Kubernetes Cloud provider. [#387](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/387)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:61]: avoids the operator to overwrite -> prevents the operator from overwriting

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 11]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
+- **[Operator]** Considered Cn Replicas when calculating component hash [#392](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/392)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:62]: I couldn't find such an item in the Chinese release notes.

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 12]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
+- **[Helm Chart]** The typo in `storageSpec` in [starrockscluster.yaml](https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/helm-charts/charts/kube-starrocks/charts/starrocks/templates/starrockscluster.yaml) is correctified. [#385](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/385)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:63]: correctified -> corrected
I don't think "correctify" is a popularly acceptable word. Use "correct" which can serve as a verb too.

File: docs/en/ecosystem_release/starrocks-kubernetes-operator.md [hunk 13]
```diff
@@ -40,9 +40,62 @@ For example, the download URL for kube-starrocks chart v1.8.6 is:
...
+
+#### 1.9.0
+
+**New features**
+
+- Add StarRocksWarehouse CRD to support StarRocks Warehouse. Note that StarRocks Warehouse is currently a feature of the StarRocks Enterprise Edition. [#323](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/323)
+
+**Enhancements**
+
+- Add the `status.``reason` field in StarRocksCluster CRD. When the apply operation of a subcontroller fails during cluster deployment, you can execute `kubectl get starrockscluster <name_of_the_starrocks_cluster_object> -oyaml`,  and view error logs shown in `status.reason` field in the returned result. [#359](https://github.com/StarRocks/starrocks-kubernetes-operator/pull/359)
```
Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:73]: image
Remove the double backticks ``

Comment (amber-create) [inline docs/en/ecosystem_release/starrocks-kubernetes-operator.md:73]: image
remove extra space

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Pull request merged**